tests/unit/conftest.py

```diff
@@ -17,3 +21,35 @@ def golang_repo_path(data_dir: Path, tmp_path: Path) -> Path:
         tar.extractall(tmp_path)
 
     return tmp_path / "golang_git_repo"
+
+
+@pytest.fixture
+def mock_async_download_binary_file() -> MagicMock:
+    async def mock_download_binary_file(
+        session: aiohttp_retry.RetryClient,
+        url: str,
+        download_path: str,
+    ) -> dict[str, str]:
+        # Simulate a file download by sleeping for a random duration
+        await asyncio.sleep(random.uniform(0.1, 0.5))
+
+        # Write some dummy data to the download path
+        with open(download_path, "wb") as file:
+            file.write(b"Mock file content")
+
+        # Return a dummy response indicating success
+        return {"status": "success", "url": url, "download_path": download_path}
+
+    return MagicMock(side_effect=mock_download_binary_file)
+
+
+class MockReadChunk:
+    def __init__(self) -> None:
+        """Create a call count."""
+        self.call_count = 0
+
+    async def read_chunk(self, size: int) -> bytes:
+        """Return a non-empty chunk for the first and second call, then an empty chunk."""
+        self.call_count += 1
+        chunks = {1: b"first_chunk-", 2: b"second_chunk-"}
+        return chunks.get(self.call_count, b"")
```

Comment:
I don't think these two should ever be needed outside the aiohttp unit tests in test_general

They would probably make more sense in that file